lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -831,26 +830,38 @@ def make_mpiexec_hydra_options(self):
             else:
                 self.log.debug("make_mpiexec_hydra_options: no rmk from HYDRA_RMK %s and hydra_info %s",
                                self.HYDRA_RMK, self.hydra_info)
+
+        default_launcher = getattr(self, 'HYDRA_LAUNCHER', None)
+        if self.options.launcher:
+            launcher = [self.options.launcher]
+            launcher.extend(default_launcher)
```

Comment:
I wouldn't use the default launcher as a fallback; if `--launcher` is specified, use that, and nothing else

don't try to outwit the user making a typo here, there *may* be a good reason to specify an unknown launcher